coreax/metrics.py

```diff
@@ -129,3 +132,141 @@ def compute(
             self.precision_threshold,
         )
         return jnp.sqrt(squared_mmd_threshold_applied)
+
+
+class CMMD(Metric):
```

Comment:
Missing the specification of the generic. I think the `MMD` metric is wrong with its specification, all `_Data` should be replaced with `Data` in `MMD`. Similarly, `_SupervisedData` with `SupervisedData` in CMMD, unless there is a requirement to maintain a TypeVar?

Reply:
Fixed

Reply:
Sorry, I should have made it clearer. The  `class MMD(Metric[_Data])` should also be to `MMD(Metric[Data])`, similarly for the supervised data cases. This means you can remove the `_SupervisedData` type var completely.

Reply:
Fixed properly now! See above for issues with the Hadamard change